src/pkg.ts

```diff
@@ -1,9 +1,20 @@
-import { readFileSync } from 'node:fs';
+import { readFileSync, existsSync } from 'node:fs';
+import { dirname, join, resolve } from 'node:path';
 import { fileURLToPath } from 'node:url';
 
-export default JSON.parse(
-  readFileSync(
-    fileURLToPath(new URL('./package.json', import.meta.url)),
-    'utf-8',
-  ),
-);
+const __dirname = dirname(fileURLToPath(import.meta.url));
+
+const currPath = join(__dirname, 'package.json');
+const parentPath = resolve(__dirname, '..', 'package.json');
+
+const pkgJsonPath = existsSync(currPath)
+  ? currPath
+  : existsSync(parentPath)
+    ? parentPath
+    : undefined;
+
+if (!pkgJsonPath) {
+  throw new Error('package.json not found');
+}
+
+export default JSON.parse(readFileSync(pkgJsonPath, 'utf-8'));
```

Comment:
This is the part I'm least happy with here.

Honestly not entirely sure why we had 2 different scenarios (something to do with publishing the `dist` dir instead of the actual package's dir?

@duskpoet maybe you remember? we discussed it many months ago 😅

Reply:
I bet there is a library for getting closest `package.json`

Reply:
We publish from dist dir, yea, but package.json is copied there, so resolve for package.json was the same. Why did you have to make it this way?

Reply:
> I bet there is a library for getting closest `package.json`

There's [`read-package-up`](https://www.npmjs.com/package/read-package-up) but I'd rather keep it simple (only possibilities are `./package.json` and `../package.json`) or avoid the need for this entirely.

Reply:
> Why did you have to make it this way?

(I really didn't want to do this, and if we merge this as-is, I want to follow-up on this regardless)

Ok I figured this out (again) I think.

Previously (the Jest setup) we ran tests from `dist/**/*.test.js` (which is also PART of the reason we had to `build` before `test`, but still need this because we are running the built CLI still).

This also explains why we persisted the snapshots manually in root `snapshots` dir!

Now, with Vitest, we run the tests directly from their `.ts` source (much better since we'd now also get proper watch mode for tests).

For all of our tests - this doesn't matter! Since we run the CLI from its built dir, but for one test - `auth.test.ts` we `import { authFlow } from './auth';` from source and run it, no CLI invocation here.

From there, `auth.ts` imports `api.ts`, which imports `pkg.ts` which has the "wrong" assumption about the relative location to `package.json`, at least when it still lives in `src/pkg.ts` (it imports `./package.json` from `src`).

Now that I have the bigger picture - I'm actually going to mock the `pkg.ts` module for the `auth.test.ts` test (since it's the only one), but I intend to follow-up.

I'll think of a proper fix for this after this PR is merged.

Reply:
oh, great find, thanks! Maybe we should retrieve package.json from the cwd instead?

Reply:
Hmm but it won't be in CWD, wdym exactly?